### PR TITLE
FIX: Let users reset their homepage choice if custom homepage is from…

### PR DESCRIPTION
**Pull request by pmusaraj**
… a theme component

This is a followup to #26291. 

@nolosb noticed that when adding the custom homepage modifier via a theme component, the user preference was missing the "(default)" option in the dropdown, so the user couldn't reset back to the custom homepage. 

This PR fixes that issue by adding a new `enabled_theme_and_component_ids` method to the theme model and using it in the serializer. It also refactors the system spec to use the same test cases for both themes and components extending the connector.

### CODE REVIEW

File: spec/models/theme_spec.rb [hunk 1]
```diff
@@ -466,6 +466,47 @@ def transpile(html)
...
+    )
+
+    theme3.update!(enabled: false)
+
+    expect(Theme.enabled_theme_and_component_ids).to contain_exactly(theme2.id, child2.id)
+
+    # theme2.destroy
+    # theme3.destroy
+
+    # expect(Theme.enabled_theme_and_component_ids).to eq([])
```
Comment (ZogStriP) [inline spec/models/theme_spec.rb:507]: Forgot to uncomment them or are they not needed anymore?

File: app/models/theme.rb [hunk 2]
```diff
@@ -241,6 +241,18 @@ def self.user_theme_ids
     end
   end
 
+  def self.enabled_theme_and_component_ids
+    get_set_cache "enabled_theme_and_component_ids" do
+      component_ids = []
+      theme_ids = Theme.user_selectable.where(enabled: true).pluck(:id)
+      theme_ids.each do |id|
+        component_ids += Theme.find(id).child_themes.where(enabled: true).pluck(:id)
+      end
```
[the remark this answers is not in the document]
Comment (pmusaraj) [reply app/models/theme.rb:250]: Good catch, should avoid the n+1.

File: spec/models/theme_spec.rb (hunk 1 above)
Comment (pmusaraj) [reply spec/models/theme_spec.rb]: Forgot to uncomment, nice catch!

**Review by ZogStriP — APPROVED** [review]

File: app/models/theme.rb [hunk 3]
```diff
@@ -241,6 +241,19 @@ def self.user_theme_ids
...
+  def self.enabled_theme_and_component_ids
+    get_set_cache "enabled_theme_and_component_ids" do
+      theme_ids = Theme.user_selectable.where(enabled: true).pluck(:id)
+      component_ids =
+        ChildTheme
+          .where(parent_theme_id: theme_ids)
+          .joins(:child_theme)
+          .where(themes: { enabled: true })
+          .pluck(:child_theme_id)
+      (theme_ids + component_ids).uniq
```
Comment (ZogStriP) [inline app/models/theme.rb:253]: You can use the `|` operator to do `+` and `.uniq` 😉 

```suggestion
      theme_ids | component_ids
```

Comment (pmusaraj) [reply app/models/theme.rb]: Ah, nice bit of code golf ⛳, that! Thanks!

**Pull request merged**